frontend/simple/views/Invite.vue

```diff
@@ -86,105 +33,110 @@
+    ]),
+    ...mapGetters([
+      'currentGroupState',
+      'memberCount'
+    ])
+  },
+  methods: {
+    updateInvitees (payload) {
+      this.form.errorMsg = null
+      Object.assign(this.form, payload.data)
     },
     async submit () {
       try {
-        this.errorMsg = null
-        // TODO: as members are successfully invited display in a
-        // seperate invitees grid and add them to some validation for duplicate invites
-        for (let member of this.members) {
-          // We need to have the latest mailbox attribute for the user
-          const mailbox = await backend.latestHash(member.state.attributes.mailbox)
-          const sentDate = new Date().toString()
+        this.form.errorMsg = null
+        const groupId = this.currentGroupId
+        const groupName = this.currentGroupState.groupName
+
+        for (let member of this.form.invitees) {
+          let mailbox = member.state.attributes.mailbox
+          let memberName = member.state.attributes.name
 
-          // We need to post the invite to the users' mailbox contract
-          const invite = new Events.HashableMailboxPostMessage(
-            {
-              from: this.$store.getters.currentGroupState.groupName,
-              headers: [this.$store.state.currentGroupId],
-              messageType: Events.HashableMailboxPostMessage.TypeInvite,
-              sentDate
-            },
-            mailbox
-          )
-          await backend.publishLogEntry(member.state.attributes.mailbox, invite)
+          if (this.isAlreadyInvited(member)) {
+            this.form.errorMsg = template(L('Failed to invite users: member {name} is already invited'),
+              { name: memberName })
+            return
+          }
 
-          // We need to make a record of the invitation in the group's contract
-          const latest = await backend.latestHash(this.$store.state.currentGroupId)
-          const invited = new Events.HashableGroupRecordInvitation(
-            {
-              username: member.state.attributes.name,
-              inviteHash: invite.toHash(),
-              sentDate
-            },
-            latest
-          )
-          await backend.publishLogEntry(this.$store.state.currentGroupId, invited)
+          let inviteToMailbox = await sbp('proposals/invite/mailRecord/create', mailbox, groupName, groupId)
+          let inviteToGroup = await sbp('proposals/invite/groupRecord/create', inviteToMailbox.toHash(), memberName, groupId)
+
+          if (this.isProposal) {
+            let latest = await backend.latestHash(groupId)
+            let proposal = new Events.HashableGroupProposal({
+              proposal: template(
+                L('This is a Vote for {name} to become a member of {group}'),
+                { name: memberName, group: groupName }
+              ),
+              // calculate the voting threshold from the group data
+              percentage: this.currentGroupState.memberApprovalPercentage * 0.01,
```

Comment:
This `* 0.01` seems a bit unclean, and the naming convention here doesn't make sense.

Either `percentage:` must be renamed, or `memberApprovalPercentage` must be renamed, or the multiplication by `* 0.01` must be done in the UI, not here (for consistencies sake, to avoid future confusion).